### ISSUE TITLE
Providing a value of type `any` should cause type error but doesn't

### ISSUE THREAD

**Issue by schickling** [microsoft/TypeScript#46955]
### 🔎 Search Terms

tuple, type check, any, 

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/C4TwDgpgBAYglhANgEwM5QLxQNoDMEoDSEIAXFKsAE5wB2A5gDRT5LIBqAhogK4TmdaIALrZhAKHEBjAPa1KUAB4ChmKLR6JEkgPQ6KACxmbkLTnERQARj2BRkMiKloByYNLkLc5eG3RZsF1QZAFtoSioXZkUJcT0zC3RZKioIKWBEEA95O1wAJh8CNDVA4LCKaij1TURY+OTU9OyvAGZCvxLS0PDK6OFYoA)

### 💻 Code

```ts
type Fields = [fieldKey: string, fieldValue: any][]

const x: any = null

// should fail but doesn't
const f: Fields = ['some str', x]

// fails correctly
const f2: Fields = ['some str', null]

// correct
const f3: Fields = [['some str', x]]

```

### 🙁 Actual behavior

```ts
const x: any = null

// should fail but doesn't
const f: Fields = ['some str', x]
```

### 🙂 Expected behavior

Should raise a type error

**Comment by MartinJohns** [comment]
Duplicate of #19541. Used search terms: `array any in:title`
